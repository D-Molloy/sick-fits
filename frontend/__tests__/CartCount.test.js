import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CartCount from "../components/CartCount";

describe("<CartCount/>", () => {
  it("renders", () => {
    shallow(<CartCount count={10} />); //dont need an expect because if it doesn't render the test will fail
  });

  it("matches the snapshot", () => {
    const wrapper = shallow(<CartCount count={11} />);
    // can take snapshots of multiple pieces using .find, etc
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("updates via props", () => {
    const wrapper = shallow(<CartCount count={11} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // MOUNT (below) SHALLOW (above)
  it("updates via props", () => {
    // mounting gives real html elements
    const wrapper = mount(<CartCount count={11} />);
    // console.log(wrapper.debug());

    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
