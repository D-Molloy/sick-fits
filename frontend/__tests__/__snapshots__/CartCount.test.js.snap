// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> matches the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        11
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount/> updates via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        11
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount/> updates via props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount/> updates via props 3`] = `
<CartCount
  count={11}
>
  <CartCount__AnimationStyles>
    <span
      className="CartCount__AnimationStyles-xxvp4g-0 hlznbU"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <div>
          <CSSTransition
            className="count"
            classNames="count"
            in={true}
            key=".$11"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <CartCount__Dot
                className="count"
              >
                <div
                  className="count CartCount__Dot-xxvp4g-1 kDZyEW"
                >
                  11
                </div>
              </CartCount__Dot>
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </span>
  </CartCount__AnimationStyles>
</CartCount>
`;

exports[`<CartCount/> updates via props 4`] = `
<CartCount
  count={10}
>
  <CartCount__AnimationStyles>
    <span
      className="CartCount__AnimationStyles-xxvp4g-0 hlznbU"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <div>
          <CSSTransition
            className="count"
            classNames="count"
            in={true}
            key=".$10"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <CartCount__Dot
                className="count"
              >
                <div
                  className="count CartCount__Dot-xxvp4g-1 kDZyEW"
                >
                  10
                </div>
              </CartCount__Dot>
            </Transition>
          </CSSTransition>
          <CSSTransition
            className="count"
            classNames="count"
            in={false}
            key=".$11"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <CartCount__Dot
                className="count"
              >
                <div
                  className="count CartCount__Dot-xxvp4g-1 kDZyEW"
                >
                  11
                </div>
              </CartCount__Dot>
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </span>
  </CartCount__AnimationStyles>
</CartCount>
`;
